x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/filters/route.ts

```diff
@@ -84,7 +85,7 @@ export const getRuleManagementFilters = (router: SecuritySolutionPluginRouter) =
         if (validationError != null) {
           return siemResponse.error({ statusCode: 500, body: validationError });
         } else {
-          return response.ok({ body: validatedBody ?? {} });
+          return response.ok({ body: validatedBody });
```

Comment:
It's interesting that you specified the return type as `Promise<IKibanaResponse<GetRuleManagementFiltersResponse>>`, but we're still able to return a `siemResponse.error()` from the handler. Does it match the `IKibanaResponse` type?

The generic type parameters for the `.ok` and `.error` methods appear to be different:

<img width="726" alt="Screenshot 2023-07-25 at 23 39 51" src="https://github.com/elastic/kibana/assets/7359339/1880d0a7-4ea8-4c71-bf0e-f992452764ab">

<img width="1419" alt="Screenshot 2023-07-25 at 23 40 02" src="https://github.com/elastic/kibana/assets/7359339/19dadc08-aa21-4749-b3dd-ce71862bb12b">